src/devices/cpu/pic16x8x/pic16x8x.cpp

```diff
@@ -0,0 +1,1443 @@
+	}
+}
+
+u8 pic16x8x_device::trisb_r()
+{
+	return m_port_tris[PORTB];
+}
+
+void pic16x8x_device::trisb_w(u8 data)
+{
+	if (m_port_tris[PORTB] != data)
+	{
+		m_port_tris[PORTB] = data;
+		m_write_port[PORTB](PORTB, m_port_data[PORTB] & u8(~m_port_tris[PORTB]), u8(~m_port_tris[PORTB]));
+	}
+}
+
+u8 pic16x8x_device::eecon1_r()
+{
+	return  m_EECON1;
+}
+
+void pic16x8x_device::eecon1_w(u8 data)
+{
+	if (!((data & WREN_FLAG) && EEWR))  // The WR bit will be inhibited from being set unless the WREN bit is set.(from datasheet)
+		CLR(data, EEWR);                // Clear the flag if it don't meet the requirements
+
+	m_EECON1 = data;                    // update register
+
+	if (EERD)                           // cheack flag condition to read eeprom
+	{
+		m_EEDATA = M_RDEEPROM(m_EEADR); // do eeprom read
+		CLR(m_EECON1, EERD_FLAG);       // clear read enable flag
+	}
+
+	if (WREN && EEWR && (m_EECON2 == 0xff)) // check flags conditions to write eeprom
+	{
+		M_WREEPROM(m_EEADR, data);          // do eeprom write
+		CLR(m_EECON1, EEWR_FLAG);           // clear write enable flag
+		SET(m_EECON1, EEIF_FLAG);           // assuming write was done, set EEIF
+	}
+}
+
+u8 pic16x8x_device::eecon2_r()
+{
+	return m_EECON2;
+}
+
+void pic16x8x_device::eecon2_w(u8 data)
+{
+	if((m_EECON2!=0x55) & (data==0x55))
+		m_EECON2 = data;
+	else if((m_EECON2==0x55) & (data==0xAA))
+		m_EECON2 = 0xff;
+	else
+		m_EECON2 = data;
+}
+
+u8 pic16x8x_device::option_r()
+{
+	return m_OPTION;
+}
+
+void pic16x8x_device::option_w(u8 data)
+{
+	m_OPTION = data;
+}
+
+/************************************************************************
+ *  Emulate the Instructions
+ ************************************************************************/
+
+void pic16x8x_device::illegal()
+{
+	logerror("PIC16x8x: PC=%03x, Illegal opcode = %04x\n", m_PREVPC, m_opcode.w);
+}
+
+/*
+    Note:
+    According to the manual, if the STATUS register is the destination for an instruction that affects the Z, DC or C bits
+    then the write to these three bits is disabled. These bits are set or cleared according to the device logic.
+    To ensure this is correctly emulated, in instructions that write to the file registers, always change the status flags
+    *after* storing the result of the instruction.
+    e.g. CALCULATE_*, SET(STATUS,*_FLAG) and CLR(STATUS,*_FLAG) should appear as the last steps of the instruction emulation.
+*/
+
+void pic16x8x_device::addlw()
+{
+	// New
+	m_ALU = m_opcode.b.l + m_W;
+	m_W = m_ALU;
+	calc_zero_flag();
+}
+
+void pic16x8x_device::addwf()
+{
+	u8 augend = get_regfile(ADDR);
+	m_ALU = augend + m_W;
+	store_result(ADDR, m_ALU);
+	calc_add_flags(augend);
+}
+
+void pic16x8x_device::andlw()
+{
+	m_ALU = m_opcode.b.l & m_W;
+	m_W = m_ALU;
+	calc_zero_flag();
+}
+
+void pic16x8x_device::andwf()
+{
+	m_ALU = get_regfile(ADDR) & m_W;
+	store_result(ADDR, m_ALU);
+	calc_zero_flag();
+	if(ADDR == 0x0b)
+		logerror("ANDWF -> INTCON - data:%x\n", m_ALU);

```

Comment:
This seems to be a leftover from debugging. If you're running MAME with `-debug`, you can use `wpdset 0x0b, 1, w` to catch writes to this register.

Reply:
You're right. Will delete this and other two following logerrors calls.

Reply:
Done. Deleted all leftovers.